### ISSUE TITLE
The plugin doesn't automatically reconnect upon network disruption.

### ISSUE THREAD

**Issue by fangfufu** [jgeboski/purple-facebook#143]
I am on 6a0a79182ebc, I am experiencing an issue similar #35 again. I have to manually disable the plugin and re-enable it to reconnect.

**Comment by jgeboski** [comment]
Is this somehow related to #89?

**Comment by fangfufu** [comment]
No, it is not. The plugin isn't disabled after network disruption.

**Comment by jgeboski** [comment]
Then it should be reconnecting after a timeout (between 8 seconds to a minute)?

**Comment by fangfufu** [comment]
It doesn't reconnect even after an hour or so. Pidgin's tray icon shows the disconnected sign. When I open up Pidgin's contact list, the status icon is spinning. My XMPP accounts are already reconnected.

**Comment by jgeboski** [comment]
What reason does it give for the disconnection?

**Comment by fangfufu** [comment]
It doesn't give any reason for disconnection. It seems to have got stuck in some sort of loop of trying to reconnect.

**Comment by jgeboski** [comment]
Does the [debugging log](https://github.com/jgeboski/purple-facebook/wiki/Debugging-Logs) show anything enlightening?

**Comment by fangfufu** [comment]
I will get onto it and report back later.

**Comment by kaberett** [comment]
(This looks like behaviour similar to what I was erroneously tacking on to comments in #89.)

**Comment by fangfufu** [comment]
@kaberett, I commented on that issue as well, then I realised that it is something different.

**Comment by fangfufu** [comment]
This is what my pidgin looks like after suspension. I have emailed you the debug log separately. I have to either restart pidgin or restart the plugin to reconnect back to Facebook. 

![pidgin](https://cloud.githubusercontent.com/assets/2323403/10117151/65eb0fa0-6445-11e5-8fb5-3ea85dc14480.png)

**Comment by fangfufu** [comment]
Perhaps purple-facebook never informed Pidgin that it had failed to connect? I don't know.

**Comment by jgeboski** [comment]
After looking at your log, this is due to libpurple failing to resolve domain names. NetworkManager sends the `NM_STATE_ASLEEP (10)` as expected, and when the system resumes, `NM_STATE_CONNECTED_LOCAL (50)`. This is where things become problematic, as an external network has yet to be activated. It would be ideal if the plugin could hint at pidgin to wait for `NM_STATE_CONNECTED_GLOBAL (70)` before reconnecting.

The account does not appear to disconnect with an error. If it did, it would end up reconnecting on its own after a period of time. But my guess is, Pidgin nullifies all errors while the account is in the re-connection phase.  And as soon as the plugin hits and error, it bails out, and propagates an error message to the user. I will have to look into this when I get time.

```
(11:53:08) network: Got StateChange from NetworkManager: 10.
...
(11:53:17) network: Got StateChange from NetworkManager: 50.
...
(11:53:17) dnsquery: Performing DNS lookup for graph.facebook.com
...
(11:53:17) dns: Got response for 'graph.facebook.com'
(11:53:17) dnsquery: Error resolving graph.facebook.com:
Name or service not known
(11:53:17) proxy: Connection attempt failed: Error resolving graph.facebook.com:
Name or service not known
(11:53:17) http: error: Unable to connect to graph.facebook.com: SSL Connection Failed
(11:53:17) http: Cancelling connection 0x7fdfbe520400...
(11:53:17) http: releasing a socket: 0x7fdfbda81500
(11:53:17) socket: invalid state: 3 (should be: 2)
(11:53:17) http: destroying socket: 0x7fdfbda81500
(11:53:17) http: Request 0x7fdfbe520400 performed without success.
...
(11:53:26) network: Got StateChange from NetworkManager: 70.
```

**Comment by fangfufu** [comment]
I think this bug was introduced in the recent revision. I didn't have this problem at one point. I wonder if this is related to #89.

**Comment by Ghabry** [comment]
Actually I can't connect to any spectrum2 network with pidgin (hangs at "Connecting").
Any other client I tried (jitsi, Psi, Trillian) worked.

**Comment by fangfufu** [comment]
Okay, I switched away from Pidgin to KDE Telepathy, which can use libpurple as a backend, it magically fixed the problem. I don't know if this helps anyone.

**Comment by ZenithDK** [comment]
I have this problem on Windows - but I'm probably using an old version of the plugin, will try a newer one and test more and if it is still an issue I will get the logs for it.

**Comment by ZenithDK** [comment]
[facebook-not-reconnecting.txt](https://github.com/jgeboski/purple-facebook/files/37641/facebook-not-reconnecting.txt)
I've attached a log I grabbed some time ago - meant to sanitize it but I'm crossing my fingers nothing sensitive is in there. Seems like passwords have already been stripped out.

**Comment by dequis** [comment]
@ZenithDK this log doesn't have anything at all. It's mostly stuff from your unrelated jabber and oscar accounts. Follow the instructions in https://github.com/jgeboski/purple-facebook/wiki/Debugging-Logs - this plugin won't log anything unless you enable UNSAFE/VERBOSE debug and that's way harder to sanitize, so don't post it publicly.

**Comment by jgeboski** [comment]
This has been resolved as of [50bb40c](https://bitbucket.org/pidgin/main/commits/50bb40c42245618dc0a2cc92901905c701ad9f58) and is in the latest release.
